Removing trading pair dependency

TheLedgerOfJoudi opened the pull request:
I moved the draft PR #801 here with the correct branching out of develop. This PR holds a better descriptive name, as well, as I added more traits to resolve the clear dependency.

wer1st commented on pallets/multicollateral-bonding-curve-pool/src/lib.rs:
```diff
@@ -873,14 +872,13 @@ impl<T: Config> Pallet<T> {
             );
             T::PriceToolsPallet::register_asset(&collateral_asset_id)?;
-            T::EnsureTradingPairExists::ensure_trading_pair_exists(
+            <T as pallet::Config>::EnsureTradingPairExists::ensure_trading_pair_exists(
```
btw you can use just `T::EnsureTradingPairExists`

wer1st commented on pallets/liquidity-proxy/src/lib.rs:
```diff
@@ -2197,11 +2193,8 @@ pub mod pallet {
-    // TODO: #441 use TradingPairSourceManager instead of trading-pair pallet
     #[pallet::config]
-    pub trait Config:
-        frame_system::Config + common::Config + assets::Config + trading_pair::Config
```
You also should to remove `trading_pair` dependency from the `Cargo.toml` files. In all pallets.

wer1st commented on pallets/pool-xyk/src/lib.rs:
```diff
@@ -307,7 +307,7 @@ impl<T: Config> Pallet<T> {
         output_b_min: Balance,
     ) -> DispatchResult {
-        let dex_info = T::DexInfoProvider::get_dex_info(&dex_id)?;
+        let dex_info = <T as pallet::Config>::DexInfoProvider::get_dex_info(&dex_id)?;
```
you can leave `T::`

wer1st commented on pallets/pool-xyk/src/utils.rs:
```diff
@@ -92,7 +91,7 @@ impl<T: Config> Pallet<T> {
         asset_b: T::AssetId,
     ) -> Result<(common::TradingPair<TechAssetIdOf<T>>, TechAccountIdOf<T>), DispatchError> {
-        let dexinfo = T::DexInfoProvider::get_dex_info(&dex_id)?;
+        let dexinfo = <T as pallet::Config>::DexInfoProvider::get_dex_info(&dex_id)?;
```
you can leave `T::`

wer1st commented:
You missed the important part - the removing of build dependecies in `Cargo.toml`

TheLedgerOfJoudi commented:
Removed trading_pair::Config as a supertrait in src/lib.rs from all pallets.
Removed trading_pair as a build dependency from pallets which do not require it in mock initializations.
T::Trait is requiring fully qualified path for disambiguation because the type could derive from multiple pallet configs.
